homeassistant/components/modbus/sensor.py

```diff
@@ -142,11 +142,12 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        regsize = entry.get(CONF_COUNT, 1) * 2
```

Comment:
regisize is a misleading variable name. regsize is fixed 2, please change the name to something like "bytecount" or "bufsize".